code/game/objects/items/weapons/cane.dm

```diff
@@ -53,4 +53,70 @@
 	else
 		SetName("cane shaft")
 		icon_state = "nullrod"
-		item_state = "foldcane"
\ No newline at end of file
+		item_state = "foldcane"
+
+/obj/item/weapon/cane/crutch
+	name ="crutch"
+	desc = "A long stick with a crosspiece at the top, used to help with walking."
+	icon_state = "crutch"
+	item_state = "crutch"
+
+/obj/item/weapon/cane/white
+	name = "white cane"
+	desc = "A white cane. They are commonly used by the blind or visually impaired as a mobility tool or as a courtesy to others."
+	icon_state = "whitecane"
+	item_state = "whitecane"
+
+/obj/item/weapon/cane/white/attack(mob/M as mob, mob/user as mob)
+    if(user.a_intent == I_HELP)
+        user.visible_message("<span class='notice'>\The [user] has lightly tapped [M] on the ankle with their white cane!</span>")
+        return TRUE
+    else
+        . = ..()
+
+
+//Code for Telescopic White Cane writen by Gozulio
+
+/obj/item/weapon/cane/white/collapsible
+	name = "telescopic white cane"
+	desc = "A telescopic white cane. They are commonly used by the blind or visually impaired as a mobility tool or as a courtesy to others."
+	icon_state = "whitecane1in"
+	item_state = "whitecanein"
+	item_icons = list(
+			slot_l_hand_str = 'icons/mob/onmob/items/lefthand.dmi',
+			slot_r_hand_str = 'icons/mob/onmob/items/righthand.dmi',
+		)
```

Comment:
Travis is regarding the indentation of list content as closing bracket as "incorrect indentation", 'unindentate' by one tab.